src/Catalog/Monitoring/Validation/Test/ValidationTestRunner.cs

```diff
@@ -0,0 +1,109 @@
+
+        /// <summary>
+        /// Validates a package.
+        /// </summary>
+        /// <returns>A <see cref="ValidationTestResult"/> which contains the results of the validation.</returns>
+        Task<ValidationTestResult> Run(ValidationTestRunData data);
+    }
+    
+    /// <summary>
+    /// Abstract class with the shared functionality between all <see cref="IValidationTestRunner"/> implementations.
+    /// </summary>
+    public abstract class ValidationTestRunner : IValidationTestRunner
+    {
+        /// <summary>
+        /// The number of tries attempted before the test is considered to have failed.
+        /// </summary>
+        public const int MaxTries = 3;
+
+        protected ValidationTestRunner(
+            ResourceToTest<IPackageTimestampMetadataResource, V2ResourceType> timestampMetadataResourceV2)
+        {
+            TimestampMetadataResourceV2 = timestampMetadataResourceV2?.Resource ??
+                throw new ArgumentNullException(nameof(timestampMetadataResourceV2));
+        }
+
+        protected readonly IPackageTimestampMetadataResource TimestampMetadataResourceV2;
+
+        /// <summary>
+        /// Checks that the current batch of catalog entries contains the entry that was created from the current state of the V2 feed.
+        /// If the metadata is missing from either V2 or the catalog, validation runs anyway.
+        /// </summary>
+        public virtual async Task<bool> ShouldRun(ValidationTestRunData data)
+        {
+            var timestampV2 = await TimestampMetadataResourceV2.Get(data);
+            var timestampCatalog = await PackageTimestampMetadata.FromCatalogEntries(data.Client, data.Entries);
+
+            // A couple notes here:
+            // 1 - If the timestamp metadata in the catalog is LESS than that of the feed, we must not be looking at the latest entry that corresponds with this package, so skip the test for now.
+            // 2 - If the timestamp metadata in the catalog is EQUAL to that of the feed, we are looking at the latest catalog entry that corresponds with this package, so run the test.
+            // 3 - If the timestamp metadata in the catalog is GREATER than that of the feed, there is likely some issue--the catalog should never have newer data than the feed--so run the test.
+            // 4 - If we failed to get the timestamp metadata in either the catalog or the feed, there is likely some issue--so run the test.
+            return !timestampV2.Last.HasValue || !timestampCatalog.Last.HasValue || timestampV2.Last <= timestampCatalog.Last;
+        }
+
+        public async Task<ValidationTestResult> Run(ValidationTestRunData data)
+        {
+            var validationTestResult = new ValidationTestResult
+            {
+                ValidationTest = this,
+                Result = TestResult.Pass
+            };
+
+            var numTries = 0;
+            do
+            {
+                numTries++;
```

Comment:
> it is either a transient issue or

Let's not put things into the "transient issue" bucket until we understand what kind of issues we are talking about here. For monitoring, you should always err on the noisy side so we can fine tune later.

> it is a scenario that we missed when we wrote our test case and the additional retries will potentially provide more debugging information.

Additionally debugging information -- sure. But we won't ever look at the debugging information if we see a success at the end :)

I really don't think this is the right thing to do here. Starting with retries seems like a good way to miss little timing issues that should not be there.